### ISSUE TITLE
Git HEAD fails to build from source, error from proc_macro2 '"::libc" is not a valid Ident'

### ISSUE THREAD

**Issue by infinity0** [gnzlbg/sleef-sys#7]
~~~~
thread 'main' panicked at '"::libc" is not a valid Ident', $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-0.4.24/src/stable.rs:563:9
note: Run with `RUST_BACKTRACE=1` for a backtrace.
~~~~

**Comment by infinity0** [comment]
Same failure with proc_macro2 0.4.4 and quote 0.6.0 which is the earliest version allowed by bindgen 0.44. (proc_macro2 0.4.1 is allowed by quote 0.6.0 but the source code of bindgen 0.44 uses APIs from proc_macro2 0.4.4.)

So I am a bit unsure how you managed to bump bindgen to 0.44 in the latest commit c3cd742352d959d765b44c946e74739466896bd0.

~~~~
thread 'main' panicked at '"::libc" is not a valid Ident', $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-0.4.4/src/stable.rs:499:9
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
   1: std::sys_common::backtrace::print
   2: std::panicking::default_hook::{{closure}}
   3: std::panicking::default_hook
   4: std::panicking::rust_panic_with_hook
   5: std::panicking::continue_panic_fmt
   6: std::panicking::begin_panic_fmt
   7: proc_macro2::stable::validate_term
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-0.4.4/src/stable.rs:499
   8: proc_macro2::stable::Ident::_new
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-0.4.4/src/stable.rs:428
   9: proc_macro2::stable::Ident::new
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-0.4.4/src/stable.rs:438
  10: proc_macro2::Ident::new
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-0.4.4/src/lib.rs:740
  11: bindgen::ir::context::BindgenContext::rust_ident_raw
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.44.0/src/ir/context.rs:937
  12: bindgen::codegen::helpers::ast_ty::raw_type
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.44.0/src/codegen/helpers.rs:135
  13: <bindgen::ir::ty::Type as bindgen::codegen::TryToRustTy>::try_to_rust_ty
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.44.0/src/codegen/mod.rs:3049
  14: <bindgen::ir::item::Item as bindgen::codegen::TryToRustTy>::try_to_rust_ty
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.44.0/src/codegen/mod.rs:3006
  15: <T as bindgen::codegen::TryToRustTyOrOpaque>::try_to_rust_ty_or_opaque
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.44.0/src/codegen/mod.rs:2900
  16: <bindgen::ir::ty::Type as bindgen::codegen::CodeGenerator>::codegen
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.44.0/src/codegen/mod.rs:701
  17: <bindgen::ir::item::Item as bindgen::codegen::CodeGenerator>::codegen
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.44.0/src/codegen/mod.rs:379
  18: <bindgen::ir::module::Module as bindgen::codegen::CodeGenerator>::codegen::{{closure}}
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.44.0/src/codegen/mod.rs:401
  19: <bindgen::ir::module::Module as bindgen::codegen::CodeGenerator>::codegen
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.44.0/src/codegen/mod.rs:431
  20: <bindgen::ir::item::Item as bindgen::codegen::CodeGenerator>::codegen
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.44.0/src/codegen/mod.rs:370
  21: bindgen::codegen::codegen::{{closure}}
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.44.0/src/codegen/mod.rs:3576
  22: bindgen::ir::context::BindgenContext::gen
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.44.0/src/ir/context.rs:1239
  23: bindgen::codegen::codegen
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.44.0/src/codegen/mod.rs:3554
  24: bindgen::Bindings::generate
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.44.0/src/lib.rs:1790
  25: bindgen::Builder::generate
             at $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.44.0/src/lib.rs:1216
  26: build_script_build::main
             at ./build.rs:113
  27: std::rt::lang_start::{{closure}}
             at libstd/rt.rs:74
  28: std::panicking::try::do_call
  29: __rust_maybe_catch_panic
  30: std::rt::lang_start_internal
  31: std::rt::lang_start
             at libstd/rt.rs:74
  32: main
  33: __libc_start_main
  34: _start
~~~~

**Comment by infinity0** [comment]
The errors go away with bindgen 0.43 (714ac61854ffce81c61e0975f40d180fc08d2ef3) which uses proc_macro2 0.3.5.

**Comment by infinity0** [comment]
Ping? This is blocking us from having rust-cargo as a crate in Debian Testing.

**Comment by infinity0** [comment]
According to https://mozilla.logbot.info/rust/20181228 it looks like this is fixed by bindgen 0.46. I'll run some more tests and close this if everythi'ngs ok

**Comment by gnzlbg** [comment]
@infinity0 I have time to look at this tomorrow.

**Comment by infinity0** [comment]
updating to bindgen 0.46 fixes the issue, so no worries i'll close this.

not sure the best way for you to deal with this in this crate - since bindgen 0.43 works but 0.44, 0.45 doesn't - perhaps the bindgen authors should yank those last two versions.

**Comment by gnzlbg** [comment]
I'll update to bindgen 0.46 :)

**Comment by gnzlbg** [comment]
I've released version 0.1.2 with the fix. Please verify that it works for you.
